arangod/Aql/MaterializeExecutor.cpp

```diff
@@ -73,7 +73,11 @@ MaterializeExecutor<localDocumentId>::MaterializeExecutor(
     MaterializeExecutor<localDocumentId>::Fetcher& /*fetcher*/, Infos& infos)
     : _buffer{infos.query().resourceMonitor()},
       _trx{infos.query().newTrxContext()},
-      _readCtx{infos} {}
+      _readCtx{infos} {
+  if constexpr (localDocumentId) {
```

Comment:
If this is `false`, `_collection` is not initialized. IMO it would be good to always initialize it to a nullptr, so that we can at least easily run into issues when dereferencing it unintentionally.

Reply:
_collection doesn't exist if it's false

Reply:
```cpp
  IRS_NO_UNIQUE_ADDRESS
  irs::utils::Need<localDocumentId, PhysicalCollection*> _collection{};
```
it will be irs::utils::Empty in such case

Reply:
And anyway it's always default initialized